### PR TITLE
deployment/kubernetes: add elements for Universal Profiling

### PR DESCRIPTION
**Pull request by florianl**
## What does this PR do?

Adding additional elements to the manifest so the Universal Profiling integration can be used.

## Why is it important?

Without these additional elements the Universal Profiling integration will not work.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

### CODE REVIEW

**Review by inge4pres — COMMENTED** [review]
Similarly to https://github.com/elastic/kibana/pull/159781, we should comment all Universal Profiling options in the manifest, and let user de-comment when installing the intregration.

File: deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml [hunk 1]
```diff
@@ -62,6 +62,12 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            - name: KUBERNETES_NODE_NAME
```
Comment (gizas) [inline deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml:65]: I think that ${env.NODE_NAME} is available for you to use. Can you check if you can use it and not need this block?

File: deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml [hunk 2]
```diff
@@ -143,3 +160,10 @@ spec:
           hostPath:
             path: /var/lib/elastic-agent-managed/kube-system/state
             type: DirectoryOrCreate
+        # Mount required for Universal Profiling.
+        # If you are not using this integration, then these volumes and the corresponding
+        # mounts can be removed.
+        - name: universal-profiling-cache
```
Comment (gizas) [inline deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml:166]: Can we have those initially commented out?

Comment (florianl) [reply deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml:166]: sure - I will comment this out.

File: deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml (hunk 1 above)
Comment (florianl) [reply deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml:65]: `KUBERNETES_NODE_NAME` is a commonly used environment variable within the Elastic org: https://github.com/search?type=code&q=org%3Aelastic+KUBERNETES_NODE_NAME

And similar to other k8 related environment variables, like `KUBERNETES_SERVICE_PORT_HTTPS` or `KUBERNETES_SERVICE_HOST`, I think it is good to be more explicit and keep the `KUBERNETES_` prefix.
Would it be ok to comment it out for the begining?

Comment (MichaelKatsoulis) [reply deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml:65]: I believe  having  both NODE_NAME and KUBERNETES_NODE_NAME in the same manifest does not make sense and can look very weird to customers. Besides that KUBERNETES_NODE_NAME is commonly used, is there any other reason that you added it?

File: deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml (hunk 2 above)
Comment (florianl) [reply deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml]: done :white_check_mark:

File: deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml (hunk 1 above)
Comment (florianl) [reply deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml]: `KUBERNETES_NODE_NAME` is an essential part of detecting and working in the k8 environment for this integration. Without it, the integration will not work.

**Review by ChrsMark — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/elastic-agent-managed-kubernetes.yaml [hunk 3]
```diff
@@ -62,6 +62,12 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            # Special environment indicator for Universal Profiling.
+            # If you are using the Universal Profiling integration, please uncomment these lines before applying.
+            #- name: KUBERNETES_NODE_NAME
+            #  valueFrom:
+            #    fieldRef:
+            #      fieldPath: spec.nodeName
```
[the remark this answers is not in the document]
Comment (florianl) [reply deploy/kubernetes/elastic-agent-managed-kubernetes.yaml]: removed `KUBERNETES_NO_NAME` with https://github.com/elastic/elastic-agent/pull/2881/commits/0266ca660c143d450c2aa71cdcac420531a29a16

**Review by inge4pres — APPROVED** [review]
👍🏼

**Review by michalpristas — APPROVED** [review]

**Pull request merged**